kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSTypeReferenceImpl.kt

```diff
@@ -64,6 +66,7 @@ class KSTypeReferenceImpl(
     }
 
     override fun resolve(): KSType {
+        analyze { recordLookup(ktType, parent) }
```

Comment:
maybe move the record to the lazy delegation clause? Or put a comment here in case in the future `ktType` is touched in another place without realizing this could escape incremental recording?